Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -257,9 +258,40 @@ lemma odd_card_iff_odd_components [Finite V] : Odd (Nat.card V) ↔
     ← Set.toFinset_card, Set.toFinset_iUnion ConnectedComponent.supp]
   rw [Finset.card_biUnion
     (fun x _ y _ hxy ↦ Set.disjoint_toFinset.mpr (pairwise_disjoint_supp_connectedComponent _ hxy))]
-  simp_rw [Set.toFinset_card, ← Nat.card_eq_fintype_card]
-  rw [Nat.card_eq_fintype_card, Fintype.card_ofFinset]
-  exact (Finset.odd_sum_iff_odd_card_odd (fun x : G.ConnectedComponent ↦ Nat.card x.supp))
+  simp_rw [Set.toFinset_card, ← Nat.card_eq_fintype_card, ← Finset.coe_filter_univ,
+    Set.ncard_coe_Finset, Set.Nat.card_coe_set_eq]
+  exact (Finset.odd_sum_iff_odd_card_odd (fun x : G.ConnectedComponent ↦ x.supp.ncard))
+
+lemma ncard_odd_components_mono [Fintype V] [DecidableEq V] {G' : SimpleGraph V}
+    [DecidableRel G.Adj] (h : G ≤ G') :
+     {c : ConnectedComponent G' | Odd c.supp.ncard}.ncard
+      ≤ {c : ConnectedComponent G | Odd c.supp.ncard}.ncard := by
+  have aux (c : G'.ConnectedComponent) (hc : Odd c.supp.ncard) :
+      {c' : G.ConnectedComponent | c'.supp ⊆ c.supp ∧ Odd c'.supp.ncard}.Nonempty := by
+    refine Set.nonempty_of_ncard_ne_zero fun h' ↦ ?_
+    simpa [-Nat.card_eq_fintype_card, -Set.coe_setOf, h']
+      using (c.odd_card_supp_iff_odd_subcomponents _ h).mp hc
+  let f : {c : ConnectedComponent G' | Odd (Nat.card c.supp)} →
+      {c : ConnectedComponent G | Odd (Nat.card c.supp)} :=
+    fun ⟨c, hc⟩ ↦ ⟨(aux c hc).choose, (aux c hc).choose_spec.2⟩
+  refine Finite.card_le_of_injective f fun c c' fcc' ↦ ?_
+  simp only [Subtype.mk.injEq, f] at fcc'
+  exact Subtype.val_injective (ConnectedComponent.eq_of_common_vertex
+    ((fcc' ▸ (aux c.1 c.2).choose_spec.1) (ConnectedComponent.nonempty_supp _).some_mem)
+      ((aux c'.1 c'.2).choose_spec.1 (ConnectedComponent.nonempty_supp _).some_mem))
+
+lemma ncard_odd_components_deleteVerts_mono [Fintype V] [DecidableEq V] (G G' : SimpleGraph V)
+    [DecidableRel G.Adj] (h : G ≤ G') (u : Set V) :
+    {c : ConnectedComponent ((⊤ : Subgraph G').deleteVerts u).coe |
+      Odd c.supp.ncard}.ncard ≤ {c :
+        ConnectedComponent ((⊤ : Subgraph G).deleteVerts u).coe | Odd c.supp.ncard}.ncard := by
```

Comment:
Do we really want this lemma given that it's a specialisation of `ncard_odd_components_mono`?

Reply:
Fair enough. I've extracted `Subgraph.deleteVerts_mono'`, and then it's just the finite instance.

I don't really see how to extract that to a global instance without making it noncomputable, so I guess the idea is to inline that where I use it, unless you have a better suggestion:)